### PR TITLE
Remove JsonSerializerOptions copy in ProblemDetailsJsonOptionsSetup

### LINKED ISSUE #46143: Remove JsonSerializerOptions copy in `ProblemDetailsJsonOptionsSetup`
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

JsonSerializerOptions.AddContext won't make the options read-only anymore (https://github.com/dotnet/runtime/issues/80527) and the following code must be updated to add context to combine with the current one.

https://github.com/dotnet/aspnetcore/blob/e5238763bddd7100823751c4a4ae0220d78160aa/src/Http/Http.Extensions/src/ProblemDetailsJsonOptionsSetup.cs#L16-L22

### PR DESCRIPTION
**Pull request by brunolins16**
Fixes #46143

### CODE REVIEW

File: src/Http/Http.Extensions/src/ProblemDetailsJsonOptionsSetup.cs [hunk 1]
```diff
@@ -11,15 +11,18 @@ internal sealed class ProblemDetailsJsonOptionsSetup : IPostConfigureOptions<Jso
...
+        switch (options.SerializerOptions.TypeInfoResolver)
         {
-            if (options.SerializerOptions.IsReadOnly)
-            {
-                options.SerializerOptions = new(options.SerializerOptions);
-            }
-
-            // Combine the current resolver with our internal problem details context
-            options.SerializerOptions.TypeInfoResolver = JsonTypeInfoResolver.Combine(options.SerializerOptions.TypeInfoResolver!, ProblemDetailsJsonContext.Default);
+            case DefaultJsonTypeInfoResolver:
```
Comment (davidfowl) [inline src/Http/Http.Extensions/src/ProblemDetailsJsonOptionsSetup.cs:16]: Add a comment as to why we're doing this.

File: src/Http/Http.Extensions/test/ProblemDetailsServiceCollectionExtensionsTest.cs [hunk 2]
```diff
@@ -166,6 +164,28 @@ public void AddProblemDetails_DoesNotCombineProblemDetailsContext_WhenNullTypeIn
         Assert.Null(jsonOptions.Value.SerializerOptions.TypeInfoResolver);
     }
 
+    [Fact]
+    public void AddProblemDetails_CombineProblemDetailsContext_WhenDefaultypeInfoResolver()
```
Comment (BrennanConroy) [inline src/Http/Http.Extensions/test/ProblemDetailsServiceCollectionExtensionsTest.cs:168]:
```suggestion
    public void AddProblemDetails_CombineProblemDetailsContext_WhenDefaultTypeInfoResolver()
```

File: src/Http/Http.Extensions/src/ProblemDetailsJsonOptionsSetup.cs [hunk 3]
```diff
@@ -11,15 +11,23 @@ internal sealed class ProblemDetailsJsonOptionsSetup : IPostConfigureOptions<Jso
...
+                // In this case, the current configuration is using a reflection-based resolver
+                // and we are prepending our internal problem details context to be evaluated
+                // first.
+                options.SerializerOptions.TypeInfoResolver = JsonTypeInfoResolver.Combine(ProblemDetailsJsonContext.Default, options.SerializerOptions.TypeInfoResolver);
+                break;
+            case not null:
+                // Combine the current resolver with our internal problem details context (adding last)
+                options.SerializerOptions.AddContext<ProblemDetailsJsonContext>();
```
Comment (BrennanConroy) [inline src/Http/Http.Extensions/src/ProblemDetailsJsonOptionsSetup.cs:24]: This is broken when someone calls `AddContext` before us right? So it'll be:
`custom context` -> `default context` -> `problem details context`

**Review by captainsafia — APPROVED** [review]

**Pull request merged**